src/ident.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use keyword::Keyword;
+
+pub type EntId = u32;            // TODO: u64? Not all DB values will be representable in a u32.
+
+/// The ability to transform entity identifiers (entids) into keyword names (idents) and back again.
+pub trait Ident {
```

Comment:
This is just super early days for starting to define rudimentary types. Not used yet, but `DB` will implement this trait.